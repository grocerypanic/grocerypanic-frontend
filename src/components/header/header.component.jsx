import React from "react";

import StopIcon from "@material-ui/icons/Stop";
import AddIcon from "@material-ui/icons/Add";
import HomeIcon from "@material-ui/icons/Home";
import StoreIcon from "@material-ui/icons/Store";
import KitchenIcon from "@material-ui/icons/Kitchen";

import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";

import { Navbar, Nav, Spinner } from "react-bootstrap";
import {
  NavContainer,
  IconContainer,
  OffIconContainer,
  MenuContainer,
} from "./header.styles";

import Routes from "../../configuration/routes";

const Header = ({ history, title, create, transaction }) => {
  React.useEffect(() => {}, []);

  const navigate = (route) => {
    if (transaction) return;
    history.push(route);
  };

  return (
    <NavContainer collapseOnSelect variant="dark" fixed="top">
      <Navbar.Brand href="#home">Panic: {title}</Navbar.Brand>
      <Nav className="mr-auto"></Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.root)}>
          <MenuContainer
            route={Routes.root}
            history={history}
            data-testid="home-icon"
          >
            <HomeIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.stores)}>
          <MenuContainer
            route={Routes.stores}
            history={history}
            data-testid="stores-icon"
          >
            <StoreIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        <Nav.Item className="nav-link" onClick={() => navigate(Routes.shelves)}>
          <MenuContainer
            route={Routes.shelves}
            history={history}
            data-testid="shelves-icon"
          >
            <KitchenIcon />
          </MenuContainer>
        </Nav.Item>
      </Nav>
      <Nav>
        {create ? (
          <Nav.Item className="nav-link" onClick={create}>
            <IconContainer>
              {transaction ? (
                <Spinner size="sm" animation="border" />
              ) : (
                <AddIcon className="AddIcon" />
              )}
            </IconContainer>
          </Nav.Item>
        ) : (
          <Nav.Item className="nav-link">
            <OffIconContainer>
              <StopIcon />
            </OffIconContainer>
          </Nav.Item>
        )}
      </Nav>
    </NavContainer>
  );
};

export default withRouter(Header);

Header.propTypes = {
  create: PropTypes.func,
  title: PropTypes.string.isRequired,
  transaction: PropTypes.bool.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};
