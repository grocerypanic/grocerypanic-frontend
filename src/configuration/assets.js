const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
