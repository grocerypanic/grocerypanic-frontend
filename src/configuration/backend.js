export const SafeMethods = ["GET", "HEAD", "OPTIONS", "TRACE"];
export const ItemFilters = ["preferred_stores", "shelf"];
export const FilterTag = "filter_tag";

export const Constants = {
  csrfLocalStorage: "csrf-token",
  csrfHeaderName: "X-CSRFToken",
  genericAPIError: "API Error",
  csrfErrorMessage: "Refresh csrf and try again.",
  alreadyRegistered: "User is already registered with this e-mail address.",
  minimumQuanity: 0,
  maximumQuantity: 10000,
  minimumShelfLife: 1,
  maximumShelfLife: 365 * 3,
  defaultShelfLife: 7,
  minimumPrice: 0,
  maximumPrice: 9999,
  maximumTransactionHistory: 14,
  AnalyticsCookieName: "PanicDataConsent-" + process.env.NODE_ENV.toUpperCase(),
  duplicateObjectApiError: { non_field_errors: ["Object already exists."] },
  pageLookupParam: "page",
  pageOverrideParam: "all_results",
};

export const Providers = {
  google: "google",
  facebook: "facebook",
};

export const Paths = {
  googleLogin: "/api/v1/auth/social/google/",
  facebookLogin: "/api/v1/auth/social/facebook/",
  manageShelves: "/api/v1/shelves/",
  manageStores: "/api/v1/stores/",
  manageItems: "/api/v1/items/",
  refreshCSRF: "/api/v1/auth/csrf/",
  manageTransactions: "/api/v1/transactions/",
};

const month = 30;
const year = 365;

export const ShelfLifeConstants = [
  { name: "3 Days", id: 3 },
  { name: "1 Week", id: 7 },
  { name: "2 Weeks", id: 14 },
  { name: "3 Weeks", id: 21 },
  { name: "1 Month", id: month },
  { name: "2 Months", id: month * 2 },
  { name: "3 Months", id: month * 3 },
  { name: "6 Months", id: month * 6 },
  { name: "1 Year", id: year },
  { name: "2 Years", id: year * 2 },
  { name: "3 Years", id: year * 3 },
];
