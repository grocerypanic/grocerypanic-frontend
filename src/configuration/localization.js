import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
