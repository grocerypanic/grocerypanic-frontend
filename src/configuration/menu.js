import Routes from "./routes";

const Options = [
  { name: "All Items", location: Routes.items },
  { name: "Shelves", location: Routes.shelves },
  { name: "Stores", location: Routes.stores },
];

export default Options;
