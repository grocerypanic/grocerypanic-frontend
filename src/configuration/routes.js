const Routes = {
  root: "/",
  menu: "/menu",
  about: "/about",
  privacy: "/privacy",
  signin: "/signin",
  shelves: "/shelves",
  splash: "/",
  stores: "/stores",
  items: "/items",
  create: "/create",
  details: "/details/:id",
  goBack: "back",
};

export const External = {
  feedback: "https://www.sharedvisionsolutions.com/#contact",
  credit: "https://www.sharedvisionsolutions.com/",
};

export default Routes;
