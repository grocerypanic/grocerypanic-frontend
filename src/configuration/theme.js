export const primary = "#006600";
export const white = "#FFFFFF";
export const secondary = "#868e96";
export const tertiary = "#808080";
export const error = "#FF0000";
export const regular = "#000000";
export const success = "#5cb85c";
export const highlight = "blue";
export const selected = "red";
export const danger = "#d9534f";
export const dark = "#495057";
export const black = "#000000";
export const warning = "#ffc107";

export const ui = {
  alertTimeout: 2000,
  resizeTimeout: 1,
};

export const navbarSize = "50px";
export const tabSize = "26px";
export const itemAttributes = {
  height: 55,
  padding: 5,
  border: 10,
  spacing: 5,
  innerHeight: 40,
  placeHolderPadding: 12,
};

export const headerMobileThreshold = 480;
export const mobileThreshold = 480;

export const graph = {
  quantityLine: "rgb(2, 117, 216)",
  changeLine: "rgb(92, 184, 92)",
  topPadding: -12,
  holdingPatternHeight: "134px",
  chartContainerHeight: "150px",
  holdingPatternColor: "secondary",
  holdingPatternScale: 1,
  holdingPatternDivHeight: 20,
  holdingPatternAnimation: "grow",
};
