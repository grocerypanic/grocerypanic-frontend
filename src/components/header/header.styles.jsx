import styled from "styled-components";
import { Navbar, Nav } from "react-bootstrap";
import { primary } from "../../configuration/theme";

export const NavContainer = styled(Navbar)`
  background-color: ${primary};
`;
