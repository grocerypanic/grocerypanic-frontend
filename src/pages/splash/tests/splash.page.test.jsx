import React from "react";
import { render, cleanup } from "@testing-library/react";
import i18next from "i18next";
import matchMediaPolyfill from "mq-polyfill";

import SpashPage from "../splash.page";
import FeedBack from "../../../components/feedback/feedback.component";
import Dialogue from "../../../components/dialogue/dialogue.component";

import { mobileThreshold } from "../../../configuration/theme";

import { HeaderContext } from "../../../providers/header/header.provider";
import initialHeaderSettings from "../../../providers/header/header.initial";

import Strings from "../../../configuration/strings";

matchMediaPolyfill(window);

jest.mock("../../../components/dialogue/dialogue.component");
jest.mock("../../../components/feedback/feedback.component");
Dialogue.mockImplementation(() => <div>MockDialogue</div>);
FeedBack.mockImplementation(() => <div>MockFeedBack</div>);

const mockHeaderUpdate = jest.fn();

window.resizeTo = function resizeTo(width, height) {
  Object.assign(this, {
    innerWidth: width,
    innerHeight: height,
    outerWidth: width,
    outerHeight: height,
  }).dispatchEvent(new this.Event("resize"));
};

describe("Check the correct props are passed to ", () => {
  let utils;
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const checkHeader = () => {
    expect(mockHeaderUpdate).toBeCalledWith({
      title: "MainHeaderTitle",
      signIn: true,
    });
    expect(i18next.t("MainHeaderTitle")).toBe(Strings.MainHeaderTitle);
  };

  const renderHelper = (props) => {
    return (utils = render(
      <HeaderContext.Provider
        value={{ ...initialHeaderSettings, updateHeader: mockHeaderUpdate }}
      >
        <SpashPage />
      </HeaderContext.Provider>
    ));
  };

  afterEach(cleanup);

  describe("in mobile mode", () => {
    beforeEach(() => {
      window.resizeTo(mobileThreshold - 50, 600);
      utils = renderHelper();
    });

    it("should call the header as expected", () => {
      checkHeader();
    });

    it("should render slide 1 as expected", async () => {
      expect(utils.getByText(Strings.SplashPage.Slide1.header)).toBeTruthy();
      expect(utils.getByText(Strings.SplashPage.Slide1.text1)).toBeTruthy();
      expect(utils.getByAltText(Strings.SplashPage.Slide1.alt)).toBeTruthy();
    });

    it("should render slide 2 as expected", async () => {
      expect(utils.getByText(Strings.SplashPage.Slide2.header)).toBeTruthy();
      expect(utils.getByText(Strings.SplashPage.Slide2.text1)).toBeTruthy();
      expect(utils.getByAltText(Strings.SplashPage.Slide2.alt)).toBeTruthy();
    });

    it("should match the snapshot on file (styles)", () => {
      expect(utils.container).toMatchSnapshot();
    });
  });

  describe("in desktop mode", () => {
    beforeEach(() => {
      window.resizeTo(800, 600);
      utils = renderHelper();
    });

    it("should call the header as expected", () => {
      checkHeader();
    });

    it("should render slide 1 as expected", async () => {
      expect(utils.getByText(Strings.SplashPage.Slide1.header)).toBeTruthy();
      expect(utils.getByText(Strings.SplashPage.Slide1.text1)).toBeTruthy();
      expect(utils.getByAltText(Strings.SplashPage.Slide1.alt)).toBeTruthy();
    });

    it("should render slide 2 as expected", async () => {
      expect(utils.getByText(Strings.SplashPage.Slide2.header)).toBeTruthy();
      expect(utils.getByText(Strings.SplashPage.Slide2.text1)).toBeTruthy();
      expect(utils.getByAltText(Strings.SplashPage.Slide2.alt)).toBeTruthy();
    });

    it("should match the snapshot on file (styles)", () => {
      expect(utils.container).toMatchSnapshot();
    });
  });
});
