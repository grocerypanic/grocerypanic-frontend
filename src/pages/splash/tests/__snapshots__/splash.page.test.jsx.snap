// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the correct props are passed to  in desktop mode should match the snapshot on file (styles) 1`] = `
.c0 {
  margin-top: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  height: calc(100vh - 50px);
  text-shadow: 3px 3px 6px rgba(0,0,0,0.8),0px 0px 3px rgba(0,0,0,1);
  width: 100vw;
  overflow: hidden;
}

.c1 .img1 {
  object-fit: cover;
  -webkit-filter: brightness(25%);
  filter: brightness(25%);
}

.c1 .img2 {
  object-fit: cover;
  -webkit-filter: brightness(25%);
  filter: brightness(25%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  font-size: 1.5em;
}

<div>
  <div
    class="c0"
  >
    <div
      class="carousel slide"
      data-interval="false"
    >
      <ol
        class="carousel-indicators"
      >
        <li
          class="active"
        />
        <li />
      </ol>
      <div
        class="carousel-inner"
      >
        <div
          class="active carousel-item"
        >
          <div
            class="c1"
          >
            <div>
              <img
                alt="shopping cart"
                class="img1"
                src="/images/cart.jpg"
              />
            </div>
            <div>
              <div
                class="carousel-caption"
              >
                <h3>
                  Take the fear out of pandemic shopping...
                </h3>
                <br />
                <div
                  class="c2"
                >
                  <p>
                    Track how often you actually use kitchen and household items. Make logical choices.
                  </p>
                </div>
                <br />
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="c1"
          >
            <div>
              <img
                alt="long shopping bill"
                class="img2"
                src="/images/bill.jpg"
              />
            </div>
            <div>
              <div
                class="carousel-caption"
              >
                <h3>
                  ... and give your credit card a break.
                </h3>
                <br />
                <div
                  class="c2"
                >
                  <p>
                    Walk out of the grocery store with what you really need.
                  </p>
                </div>
                <br />
              </div>
            </div>
          </div>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Check the correct props are passed to  in mobile mode should match the snapshot on file (styles) 1`] = `
.c0 {
  margin-top: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .nav-tabs {
  height: 28px !important;
  font-size: 0.75em;
}

.c0 a[role="tab"] {
  padding-bottom: 0px;
  padding-top: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.c1 {
  height: calc(100vh - 50px);
  text-shadow: 3px 3px 6px rgba(0,0,0,0.8),0px 0px 3px rgba(0,0,0,1);
  width: 100vw;
  overflow: hidden;
}

.c1 .img1 {
  object-fit: cover;
  object-position: -350px -100px;
  -webkit-filter: brightness(25%);
  filter: brightness(25%);
}

.c1 .img2 {
  object-fit: cover;
  object-position: -350px -250px;
  -webkit-filter: brightness(25%);
  filter: brightness(25%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  font-size: 1em;
}

<div>
  <div
    class="c0"
  >
    <div
      class="carousel slide"
      data-interval="false"
    >
      <ol
        class="carousel-indicators"
      >
        <li
          class="active"
        />
        <li />
      </ol>
      <div
        class="carousel-inner"
      >
        <div
          class="active carousel-item"
        >
          <div
            class="c1"
          >
            <div>
              <img
                alt="shopping cart"
                class="img1"
                src="/images/cart.jpg"
              />
            </div>
            <div>
              <div
                class="carousel-caption"
              >
                <h3>
                  Take the fear out of pandemic shopping...
                </h3>
                <br />
                <div
                  class="c2"
                >
                  <p>
                    Track how often you actually use kitchen and household items. Make logical choices.
                  </p>
                </div>
                <br />
              </div>
            </div>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <div
            class="c1"
          >
            <div>
              <img
                alt="long shopping bill"
                class="img2"
                src="/images/bill.jpg"
              />
            </div>
            <div>
              <div
                class="carousel-caption"
              >
                <h3>
                  ... and give your credit card a break.
                </h3>
                <br />
                <div
                  class="c2"
                >
                  <p>
                    Walk out of the grocery store with what you really need.
                  </p>
                </div>
                <br />
              </div>
            </div>
          </div>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        href="#"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>
</div>
`;
